src/renderer/components/dock/pod-logs.tsx

```diff
@@ -78,39 +82,81 @@ export class PodLogs extends React.Component<Props> {
     await this.load();
   }
 
+  /**
+   * Function loads more logs (usually after user scrolls to top) and sets proper
+   * scrolling position
+   * @param scrollHeight previous scrollHeight position before adding new lines
+   */
+  preload = async (scrollHeight: number) => {
+    if (podLogsStore.lines < logRange) return;
+    this.preloading = true;
+    await podLogsStore.load(this.tabId).then(() => this.preloading = false);
+    if (this.logsElement.scrollHeight > scrollHeight) {
+      // Set scroll position back to place where preloading started
+      this.logsElement.scrollTop = this.logsElement.scrollHeight - scrollHeight - 48;
+    }
+  }
+
+  /**
+   * Computed prop which returns logs with or without timestamps added to each line and
+   * does separation between new and old logs
+   * @returns {Array} An array with 2 string items - [oldLogs, newLogs]
+   */
   @computed
   get logs() {
-    if (!podLogsStore.logs.has(this.tabId)) return;
-    const { oldLogs, newLogs } = podLogsStore.logs.get(this.tabId);
+    if (!podLogsStore.logs.has(this.tabId)) return [];
+    const logs = podLogsStore.logs.get(this.tabId);
     const { getData, removeTimestamps } = podLogsStore;
     const { showTimestamps } = getData(this.tabId);
-    return {
-      oldLogs: showTimestamps ? oldLogs : removeTimestamps(oldLogs),
-      newLogs: showTimestamps ? newLogs : removeTimestamps(newLogs)
+    let oldLogs = logs;
+    let newLogs = "";
+    if (this.newLogsSince) {
+      // Finding separator timestamp in logs
+      const index = logs.indexOf(this.newLogsSince);
```

Comment:
I think this really should be stored someplace else so that the entire logs string isn't linearly searched on every render